### PR TITLE
Batch update aws packages

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

- aws-c-io: 0.11.0 -> 0.11.2 https://github.com/awslabs/aws-c-io/releases/tag/v0.11.2
- aws-c-common: 0.7.0 -> 0.7.4 https://github.com/awslabs/aws-c-common/releases/tag/v0.7.4
- aws-c-event-stream: 0.2.7 -> 0.2.12 https://github.com/awslabs/aws-c-event-stream/releases/tag/v0.2.12
- aws-c-http: 0.6.15 -> 0.6.17 https://github.com/awslabs/aws-c-http/releases/tag/v0.6.17
- aws-c-mqtt: 0.7.10 -> 0.7.11 https://github.com/awslabs/aws-c-mqtt/releases/tag/v0.7.11
- aws-c-s3: 0.1.39 -> 0.1.43 https://github.com/awslabs/aws-c-s3/releases/tag/v0.1.43
- aws-crt-cpp: 0.17.28 -> 0.17.32 https://github.com/awslabs/aws-crt-cpp/releases/tag/v0.17.32

Closes #184790
Closes #184787

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Closes #180614
Closes #180428

### CODE REVIEW

**Comment by jiegec** [comment]
Oops, it breaks nix, let me look at this.

File: pkgs/development/libraries/aws-crt-cpp/default.nix [hunk 1]
```diff
@@ -17,30 +17,21 @@
...
 
   patches = [
     # Correct include path for split outputs.
     # https://github.com/awslabs/aws-crt-cpp/pull/325
     ./0001-build-Make-includedir-properly-overrideable.patch
-
-    # Fix build with new input stream api
-    # https://github.com/awslabs/aws-crt-cpp/pull/341
-    # Remove with next release
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/aws-crt-cpp/default.nix:39]: Please remove the unused fetchpatch from inputs.

Comment (jiegec) [reply pkgs/development/libraries/aws-crt-cpp/default.nix:39]: Done

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg build nix

**Comment by vcunat** [comment]
Note that `i686-linux` had been pinned to an old version of aws-sdk-cpp for a long time.

**Comment by vcunat** [comment]
Oh, link: https://hydra.nixos.org/build/189042801

**Comment by jiegec** [comment]
The error message seems the same as https://github.com/aws/aws-sdk-cpp/issues/1940

**Comment by jiegec** [comment]
We may have to pin aws-crt-cpp to an old version on i686-linux.

**Comment by SuperSandro2000** [comment]
With the latest version just a test fails:

```
aws-sdk-cpp> /build/source/aws-cpp-sdk-core-tests/aws/client/AdaptiveRetryStrategyTest.cpp:190: Failure
aws-sdk-cpp> The difference between step.newTokenBucketRate and retryTokenBucket.T_GetFillRate() is 0.0017265603999998547, which exceeds EPSILON, where
aws-sdk-cpp> step.newTokenBucketRate evaluates to 0.69120000000000004,
aws-sdk-cpp> retryTokenBucket.T_GetFillRate() evaluates to 0.68947343960000018, and
aws-sdk-cpp> EPSILON evaluates to 9.9999999999999995e-07.
aws-sdk-cpp> /build/source/aws-cpp-sdk-core-tests/aws/client/AdaptiveRetryStrategyTest.cpp:190: Failure
aws-sdk-cpp> The difference between step.newTokenBucketRate and retryTokenBucket.T_GetFillRate() is 0.0023503203999999389, which exceeds EPSILON, where
aws-sdk-cpp> step.newTokenBucketRate evaluates to 1.0975999999999999,
aws-sdk-cpp> retryTokenBucket.T_GetFillRate() evaluates to 1.0952496796, and
aws-sdk-cpp> EPSILON evaluates to 9.9999999999999995e-07.
aws-sdk-cpp> /build/source/aws-cpp-sdk-core-tests/aws/client/AdaptiveRetryStrategyTest.cpp:190: Failure
aws-sdk-cpp> The difference between step.newTokenBucketRate and retryTokenBucket.T_GetFillRate() is 0.0014017472944809484, which exceeds EPSILON, where
aws-sdk-cpp> step.newTokenBucketRate evaluates to 3.8218740000000002,
aws-sdk-cpp> retryTokenBucket.T_GetFillRate() evaluates to 3.8204722527055193, and
aws-sdk-cpp> EPSILON evaluates to 9.9999999999999995e-07.
aws-sdk-cpp> /build/source/aws-cpp-sdk-core-tests/aws/client/AdaptiveRetryStrategyTest.cpp:190: Failure
aws-sdk-cpp> The difference between step.newTokenBucketRate and retryTokenBucket.T_GetFillRate() is 0.00055633554627831927, which exceeds EPSILON, where
aws-sdk-cpp> step.newTokenBucketRate evaluates to 4.2003729999999999,
aws-sdk-cpp> retryTokenBucket.T_GetFillRate() evaluates to 4.1998166644537216, and
aws-sdk-cpp> EPSILON evaluates to 9.9999999999999995e-07.
aws-sdk-cpp> /build/source/aws-cpp-sdk-core-tests/aws/client/AdaptiveRetryStrategyTest.cpp:190: Failure
aws-sdk-cpp> The difference between step.newTokenBucketRate and retryTokenBucket.T_GetFillRate() is 0.0019530492986188364, which exceeds EPSILON, where
aws-sdk-cpp> step.newTokenBucketRate evaluates to 3.452226,
aws-sdk-cpp> retryTokenBucket.T_GetFillRate() evaluates to 3.4502729507013812, and
aws-sdk-cpp> EPSILON evaluates to 9.9999999999999995e-07.
```

**Comment by SuperSandro2000** [comment]
That is the same problem that we had in https://github.com/NixOS/nixpkgs/pull/169909

I don't really want to pin i686 to an older version if we need to start pin other aws libraries to an older version, too.

**Comment by SuperSandro2000** [comment]
``pkgsi686Linux.nix`` build on first try, ``pkgsi686Linux.aws-sdk-cpp`` is not building and I marked it broken when building with ec2. Not ideal but better than pinning one package right now and more in the future. #188983

**Pull request merged**